frappe/boot.py

```diff
@@ -248,6 +254,21 @@ def get_user_pages_or_reports(parent, cache=False):
 	return has_role
 
 
+def _run_with_permission_query(query: "MySQLQueryBuilder", doctype: str) -> list[dict]:
+	"""
+	Adds Permission Query (Server Script) conditions and runs/executes modified query
+	Note: Works only if 'WHERE' is the last clause in the query
+	"""
+	db_query = DatabaseQuery(doctype, frappe.session.user)
+	permission_query = db_query.get_permission_query_conditions()
+
+	query = query.get_sql()
+	if permission_query:
+		query = query + " AND " + permission_query
```

Comment:
```suggestion
	if permission_query := DatabaseQuery(doctype, frappe.session.user).get_permission_query_conditions():
		query = f"{query} AND {permission_query}"

```

Reply:
I'll make `permission_query` declaration a one-liner, but the walrus bit is hard to read personally.